beacon-chain/rpc/service.go

```diff
@@ -598,6 +611,49 @@ func (s *Service) Start() {
 	}()
 }
 
+func updateTrackValidatorCacheWithProposerSettings(ctx context.Context, chain blockchain.ChainInfoFetcher, settings *proposersettings.ProposerSettingsPayload, tackedValidatorCache *cache.TrackedValidatorsCache) error {
```

Comment:
Why do we call this function instead of just `prepareProposer`  that updates the cache?